src/permissions.ts

```diff
@@ -159,10 +160,33 @@ export function permissionMiddleware<T>(
         });
       }
       if (!data || (["update", "delete"].includes(method) && data?.__t && !req.body?.__t)) {
-        throw new APIError({
-          status: 404,
-          title: `Document ${req.params.id} not found for model ${model.modelName}`,
-        });
+        // Check if document exists but is hidden
```

Comment:
Add a //TODO: to make this general, because consumers might hide it with something other than deleted/disabled/archived